proposals/NNNN-android-manager-listener-update.md

```diff
@@ -0,0 +1,253 @@
+    Language desiredLanguage = this.hmiLanguage;
+    Language actualLanguage = getRegisterAppInterfaceResponse().getLanguage();
+    
+    if (actualLanguage != null && !desiredLanguage.equals(actualLanguage) && supportedLanguages.contains(actualLanguage)) {
+        SdlManager.Updater updater = managerListener.onConfigurationUpdate(actualLanguage);
+        
+        if (updater != null) {
+            updater.update(actualLanguage);
+        }
+    }
+    
+    // first we need to check if the SDL language matches or if we need to update configuration
+    DebugTool.logInfo("Proxy is connected. Initializing managers.");
+    initialize();
+}
+``` 
+
+### SdlManager.Updater
+
+The updater will be a nested class which behaves similar to the static `Builder` subclass.
+
+```java
+public class SdlManager {
+public class Updater {
+    private @Nullable String appName;
+    private @Nullable String shortAppName;
+    private @Nullable List<TTSChunk> ttsName;
+    private @Nullable List<String> voiceRecognitionCommandNames;
+
+    public Updater(String appName, String shortAppName, List<TTSChunk> ttsName, List<String> voiceRecognitionCommandNames) {
+        this.appName = appName;
+        this.shortAppName = shortAppName;
+        this.ttsName = ttsName;
+        this.voiceRecognitionCommandNames = voiceRecognitionCommandNames;
+        ChangeRegistration
+    }
+
+    public void setAppName(String appName) {
+        this.appName = appName;
+    }
+
+    public void setShortAppName(String shortAppName) {
+        this.shortAppName = shortAppName;
+    }
+
+    public void setTtsName(List<TTSChunk> ttsName) {
+        this.ttsName = ttsName;
+    }
+
+    public void setVoiceRecognitionCommandNames(List<String> voiceRecognitionCommandNames) {
+        this.voiceRecognitionCommandNames = voiceRecognitionCommandNames;
+    }
+
+    void update(Language language) {
+        ChangeRegistration request = new ChangeRegistration(language, language);
+        SdlManager.this.hmiLanguage = language;
+
+        if (appName != null) {
+            request.setAppName(appName);
+            SdlManager.this.appName = appName;
+        }
+
+        if (shortAppName != null) {
+            request.setNgnMediaScreenAppName(shortAppName);
+            SdlManager.this.shortAppName = shortAppName;
+        }
+
+        if (ttsName != null) {
+            request.setTtsName(ttsName);
+            SdlManager.this.ttsChunks = new Vector<>(ttsName);
+        }
+
+        if (voiceRecognitionCommandNames != null) {
+            request.setVrSynonyms(voiceRecognitionCommandNames);
+            SdlManager.this.vrSynonyms = new Vector<>(voiceRecognitionCommandNames);
+        }
+
+        SdlManager.this.sendRPC(request);
+    }
+}
+```
+
+## Potential downsides
+
+From a technical point of view there is no downside to the added features. However as Java interfaces force you to implement every single method which can be distracting. 
```

Comment:
Update to: From a technical point of view there is no downside to the **add this feature**. **However,** as Java interfaces force you to implement every single **method, it can be distracting.**